internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -101,78 +103,95 @@ func (r *ClusterDeploymentsReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, nil
 	}
 
+	clusterInstall := &hiveext.AgentClusterInstall{}
+	aciName := clusterDeployment.Spec.ClusterInstallRef.Name
+	err = r.Get(ctx,
+		types.NamespacedName{
+			Namespace: clusterDeployment.Namespace,
+			Name:      aciName,
+		},
+		clusterInstall)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			r.Log.WithField("AgentClusterInstall", aciName).Infof("AgentClusterInstall does not yet exist for ClusterDeployment %s", clusterDeployment.Name)
```

Comment:
It doesn't need to exist. we are not the only platform that works with Hive.

Reply:
Done